### PR TITLE
Disallow selecting singletons

### LINKED ISSUE #17722: Cannot select a foreign item when collection has a M2O to a singleton
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

When a collection referes to another collection with an MSO relationship, and the related collection is a singleton, the drawer of the M2O interface doesn't show the singleton item - the drawer is just blank.

https://user-images.githubusercontent.com/34009329/223357025-f9be4c4a-d6e7-47e4-9865-30a11b600bbe.mov

### To Reproduce

1. Create a collection as and mark it as singleton
2. Create another collection with a M2O relationship to the singleton collection
3. Open the collection with the related singleton collection and click on the m2o interface. 

The drawer will open but will not contain an item.
When the collection from step 1 is not a singleton, the behavior is as expected so the items will appear in the drawer and one can be selected. But when the collection is a singleton, no item is shown.

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by Nitwel**
I'm wondering what the right approach to solve this would be. We could fix the bug to support singletons in relations, but then you could end up with multiple singleton items in the database even though the app can only ever show one, which is not good at all.

What I decided to do for now is to do it as the m2a is currently doing it by disallowing to select singletons on the relation which makes sense to me as we didn't really develop relational interfaces with the intent of using it with singletons. That would probably a separate improvement to relational interfaces.

Fixes #17722

### CODE REVIEW

**Review by paescuj — APPROVED** [review]
The fix itself is fine but it has no effect if the collection is subsequently converted to singleton. In this case, could we just display a message in the (empty) drawer, for example?

**Comment by jaads** [comment]
Isn't this a common use case? I mean to have a M20 relationship to a singleton? For example to place multiple products on a landingpage (which is a singleton) as shown in the issue https://github.com/directus/directus/issues/17722? 🤔 

The problem you mentioned @Nitwel with having multiple sinlgeton items in the database is actually another issue https://github.com/directus/directus/issues/17080. I came across this bug here, when investigating on https://github.com/directus/directus/issues/17080.

**Comment by Nitwel** [comment]
This is not about if it is a usecase but rather do we currently support relations to singletons which is a big no as even the api allows for creating multiple items over relations to a singleton collection (which it shouldn't allow). That's why we should communicate clearly to the user that what he's trying isn't currently supported/implemented.

And good catch @paescuj , I will implement a banner if you have a relation to a singleton so that case is covered too!

**Comment by jaads** [comment]
The banner is working as expected 👍 

I have the feeling that this could somewhat break current applications although it should have never worked in the first place. But it actually seems like it when looking at https://github.com/directus/directus/issues/17080.

However should we create a follow up issue or discussion on this to support M2O to singletons? I'm quite surprised now that this is not possible tbh 😄

**Comment by jaads** [comment]
So https://github.com/directus/directus/issues/17722 is actually a feature request, rather then a bug, right?

File: app/src/lang/translations/en-US.yaml [hunk 1]
```diff
@@ -533,6 +533,7 @@ create_collection: Create Collection
 no_collections_copy_admin: You don’t have any Collections yet. Click the button below to get started.
 no_collections_copy: You don’t have any Collections yet. Please contact your system administrator.
 relationship_not_setup: The relationship hasn't been configured correctly
+no_singleton_relations: Relationships to Singletons are not supported
```
Comment (jaads) [inline app/src/lang/translations/en-US.yaml:536]: "Singletons" should be lowercase

**Comment by jaads** [comment]
I'm still a bit uncertain about this. @rijkvanzanten do you have any thoughts on this?

**Comment by rijkvanzanten** [comment]
> I'm still a bit uncertain about this. @rijkvanzanten do you have any thoughts on this?

I think this is an acceptable solution _for now_ as it prevents a footgun. I think this raises a way bigger question around what Singletons _are_, what we want them to be, and how they should behave and interact with relations. For example, another point of confusion that folk have expressed before is how singletons behave with permissions. For example, if you don't have permissions to read item `1`, should it still perform everything the same but with item `2`? How do you control what role gets to see what single item? Is it still a single item in that case? Etc

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**